chromadb/db/pg.py

```diff
@@ -0,0 +1,544 @@
+        cursor.close()
+        
+
+    def count(self, collection_name: str):
+        collection_uuid = self.get_collection_uuid_from_name(collection_name)
+        return self_count(collection_uuid)
+
+    def _count(self, collection_uuid: str):
+        query = f"SELECT COUNT(*) FROM embeddings WHERE collection_uuid = %s"
+        params = (collection_uuid,)
+
+        cursor = self._run_query(query, *params)
+        row = cursor.fetchone()
+        if row:
+            return row[0]
+        else:
+            return 0
+    
+    def delete(
+        self,
+        where: Where = {},
+        collection_name: Optional[str] = None,
+        collection_uuid: Optional[str] = None,
+        ids: Optional[IDs] = None,
+        where_document: WhereDocument = {},
+    ):
+        if collection_name == None and collection_uuid == None:
+            raise TypeError("Arguments collection_name and collection_uuid cannot both be None")
+
+        if collection_name is not None:
+            collection_uuid = self.get_collection_uuid_from_name(collection_name)
+
+        where_str = self._create_where_clause(
+            # collection_uuid must be defined at this point, cast it for typechecker
+            cast(str, collection_uuid),
+            ids=ids,
+            where=where,
+            where_document=where_document,
+        )
+
+        return self._delete(where_str)
+
+    def _delete(self):
+        query = f"DELETE FROM embeddings {where_str} RETURNING uuid;"
+        cursor = self._run_query(query)
+        deleted_uuids = [row[0] for row in cursor.fetchall()]
+        cursor.close()
+        return deleted_uuids
+        
+
+    def reset(self):
+        query = "DROP TABLE collections"
+        cursor = self._run_query(query)
+        cursor.close()
+        
+        query = "DROP TABLE embeddings"
+        cursor = self._run_query(query)
+        cursor.close()
+        
+        self._initialize_db()
+
+    def _get_index_dims(self, collection_name: str):
+        query = "SELECT vector_dims(embedding) FROM embeddings JOIN collections on collections.uuid = embeddings.collection_uuid WHERE collections.name = %s"
+        params = (collection_name,)
+        return self._run_query(query, params).fetchone()[0]
+
+    
+    def _get_doc_length(self, collection_name):
+        query = "SELECT COUNT(DISTINCT(id)) FROM embeddings JOIN collections on collections.uuid = embeddings.collection_uuid WHERE collections.name = %s"
+        params = (collection_name,)
+        return self._run_query(query, params).fetchone()[0]
+
+    def _generate_embedding_distance_select(self, embeddings):
+        distances = []
+        placeholders = []
+        for embedding in embeddings:
+            distances.append("SELECT uuid, embedding <-> %s AS distance FROM embeddings WHERE collection_uuid = %s")
+            placeholders.append(str(embedding))
+        query = ' UNION ALL '.join(distances)
+        return query, placeholders
```

Comment:
As per https://github.com/pgvector/pgvector/issues/59 -- also result set needs to be de-duped